js/ui/save-html.js

```diff
@@ -60,9 +61,51 @@ define(
 
                     }
                 ;
-                addButton("Save as HTML", self.toString(), "Overview.html", function () { self.toHTMLSource(); });
-                addButton("Save as XHTML5", self.toXML(5), "Overview.xhtml", function () { self.toXHTMLSource(5); });
-                addButton("Save as XHTML 1.0", self.toXML(1), "Overview.xhtml", function () { self.toXHTMLSource(1); });
+                var htmlToDataURL = function(data){
+                    return "data:text/html;charset=utf-8," + encodeURIComponent(data);
+                };
+
+                // HTML
+                var htmlOpts = {
+                    title: "Save as HTML",
+                    url: htmlToDataURL(self.toString()),
+                    popupContent: this.toHTMLSource.bind(this),
+                    fileName:  "Overview.html"
+                };
+                addButton(htmlOpts);
+
+                // XHTML5
+                var xhtml5Opts ={
+                    title: "Save as XHTML5",
+                    url: htmlToDataURL(self.toXML(5)),
+                    fileName: "Overview.xhtml",
+                    popupContent: function () {
+                        self.toXHTMLSource(5);
+                    },
+                };
+                addButton(xhtml5Opts);
+
+                // XHTML 1.0
+                var xhtml1Opts ={
+                    title: "Save as XHTML 1.0",
+                    url: htmlToDataURL(self.toXML(1)),
+                    fileName: "Overview.xhtml",
+                    popupContent: function () { self.toXHTMLSource(1); },
+                };
+                addButton(xhtml1Opts);
+
+                // ePub
+                var ePubOpts = {
```

Comment:
I think this makes the intention a lot clearer. I like it. 

I'd change it to the following but YMMV:

```js
 // ePub
addButton({
    title: "Save as EPUB 3",
    url: this.makeEPubHref(),
    fileName: "spec.epub",
    popupContent: function (){
        window.open(self.makeEPubHref(), "_blank");         
    },
    type: "application/epub+zip"
});
```